hls4ml/templates/vivado/nnet_utils/nnet_garnet.h

```diff
@@ -45,7 +49,11 @@ namespace nnet {
       for (unsigned iw = 1; iw < table_size; ++iw) {
         index = iw;
         distance.range(CONFIG_T::distance_width - 1, 0) = index.range(CONFIG_T::distance_width - 1, 0);
+#ifdef NO_VIVADO
+        edge_weights_table[iw] = std::exp(-distance.to_double() * distance.to_double());
```

Comment:
Is this cast to double introducing some difference between `hls::exp` (in cosim) and `std::exp` (in python API)? Could we apply some ad-hoc truncation/rounding here to make them agree?